libraries/aws_ec2_instance.rb

```diff
@@ -64,14 +63,15 @@ def security_groups
         end
 
         def tags
+          return [] unless @instance[:tags]
           @instance[:tags].map { |tag| { key: tag[:key], value: tag[:value] } }
         end
       end
     end
   end
 
   def exists?
-    !@instance.nil? && !@instance.empty?
+    failed_resource? ? false : true
```

Comment:
could this not just be `unless failed_resource?`

Reply:
Actually just `!failed_resource?` would be enough now.
This was before adding default value to the `@failed_resource` instance variable.